crates/trie/trie/src/proof_v2/mod.rs

```diff
@@ -99,13 +115,219 @@ where
             .unwrap_or_else(|| Vec::with_capacity(16))
     }
 
+    /// Returns true if the proof of a node at the given path should be retained. This may move the
```

Comment:
maybe we can add a tldr on _why_ the proof should be retained so its easy to follow _how_ it works

e.g
```suggestion
    /// Returns true if the proof of a node at the given path should be retained. 
    /// A node is retained if its path is a prefix of any target.
    /// This may move the
```